src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -317,6 +316,22 @@ private async Task<Recommendation> GetSelectedRecommendationFromPreviousDeployme
             return selectedRecommendation;
         }
 
+        private Recommendation? GetRecommendationForRedeployment(List<Recommendation> recommendations, CloudApplication deployedApplication, string deploymentProjectPath)
+        {
+            if (!string.IsNullOrEmpty(deploymentProjectPath))
+            {
+                // There will only be ONE available recommendation if the deployment project path is provided. 
+                var recommendation = recommendations.First();
+                if (!string.Equals(recommendation.Recipe.BaseRecipeId, deployedApplication.RecipeId, StringComparison.Ordinal))

```

Comment:
You should also check if `recommendation.Recipe.BaseRecipeId` is also equal in case the user passes a deployment project for an application that is already using that project. Which gets back to my original comment of needing the double check. I suspose here you could just call your `IsCompatible` method.

Reply:
I have reverted a the `IsCompatible` function to just return a boolean. I have added a separate private method called `GetRecommendationForRedeployment` in the `DeployCommand` which performs this double check.